ui/lib/kv/addon/components/kv-secret-form.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency';
+import { inject as service } from '@ember/service';
+import KVObject from 'vault/lib/kv-object';
+
+/**
+ * @module KvSecretForm is used for creating a new secret or secret version (also considered 'editing')
+ *
+ * <KvSecretForm
+ *  @secret={{@secret}}
+ *  @onSave={{transition-to "vault.cluster.secrets.backend.kv.secret.details" @secret.path}}
+ *  @onCancel={{transition-to "vault.cluster.secrets.backend.kv.list"}}
+ * />
+ *
+ * @param {model} secret - Ember data model: 'kv/data'
+ * @param {callback} onSave - callback (usually a transition) from parent to perform after the model is saved
+ * @param {callback} onCancel - callback (usually a transition) from parent to perform when cancel button is clicked
+ */
+
+export default class KvSecretForm extends Component {
+  @service flashMessages;
+  @tracked showJsonView = false;
+  @tracked errorMessage;
+  @tracked modelValidations;
+  @tracked invalidFormAlert;
+
+  get emptyJson() {
+    // if secretData is null, this specially formats a blank object and renders a nice initial state for the json editor
+    return KVObject.create({ content: [{ name: '', value: '' }] }).toJSONString(true);
```

Comment:
It took me forever to track down how this was originally happening. First `secretData` is made a KV object in the parent component `SecretEdit` [here](https://github.com/hashicorp/vault/blob/e55c18ed1299e0d36b88e603fa9f12adaf8e75dc/ui/app/components/secret-edit.js#L48)

And then `addRow()` is called when initializing the json editor and empty key value pairs are pushed into the object [here](https://github.com/hashicorp/vault/blob/96f72891fbdb8d6fa861320775d81a6d7020b481/ui/app/components/secret-create-or-update.js#L207-L210)

I wanted to preserve the initial state view but felt like it could be simplified by doing it this way. (Hopefully this doesn't over-simplify it too much...)